src/library/autodj/trackselector.h

```diff
@@ -0,0 +1,94 @@
+#ifndef TRACKSELECTOR_H
+#define TRACKSELECTOR_H
+
+#include <QObject>
+
+#include "library/playlisttablemodel.h"
+#include "track/track.h"
+#include "util/class.h"
+
+
+class ITrackSelector
+{
+  public:
+    virtual ~ITrackSelector(){} // do not forget this
+
+    virtual const QString getName() = 0;
+
+    virtual const QString getId() = 0;
+
+    virtual TrackPointer getNextTrackFromQueue() = 0;
+
+
+};
+
+Q_DECLARE_INTERFACE(ITrackSelector, "ITrackSelector") // define this out of namespace scope
+
+
+/*!
+    \class TrackSelector
+    \brief Base class that is responsible for selecting the next AutoDJ tack
+*/
+
+class TrackSelector : public QObject, public ITrackSelector {
+    Q_OBJECT
+    Q_INTERFACES(ITrackSelector)
+  public:
+    TrackSelector(AutoDJProcessor* processor);
```

Comment:
This violates the rule, that plain pointers must not be stored. Can't we pass the pointer when needed to getNextTrackFromQueue?